src/deepsparse/transformers/pipelines.py

```diff
@@ -0,0 +1,804 @@
+            f"are {SUPPORTED_ENGINES}"
+        )
+
+    task_info = SUPPORTED_TASKS[task]
+    model_path = model_path or _get_default_model_path(task_info)
+    model_name = model_name or task_info.default_model_name
+
+    # default the tokenizer and config to given model name
+    tokenizer = tokenizer or model_name
+    config = config or model_name
+
+    # create model
+    if model_path.startswith("zoo:"):
+        model_path = _download_zoo_model(model_path)
+    model, input_names = _create_model(model_path, engine_type, num_cores, num_sockets)
+
+    # Instantiate tokenizer if needed
+    if isinstance(tokenizer, (str, tuple)):
+        if isinstance(tokenizer, tuple):
+            # For tuple we have (tokenizer name, {kwargs})
+            tokenizer = AutoTokenizer.from_pretrained(tokenizer[0], **tokenizer[1])
+        else:
+            tokenizer = AutoTokenizer.from_pretrained(tokenizer, use_fast=False)
+
+    # Instantiate config if needed
+    if config is not None and isinstance(config, str):
+        config = AutoConfig.from_pretrained(config)
+
+    return task_info.pipeline_constructor(
+        model=model,
+        tokenizer=tokenizer,
+        config=config,
+        engine_type=engine_type,
+        max_length=max_length,
+        input_names=input_names,
+        **kwargs,
+    )
+
+
+def _get_default_model_path(task_info: TaskInfo) -> str:
+    if cpu.cpu_vnni_compatible() and task_info.default_quant_stub:
+        return task_info.default_quant_stub
+    return task_info.default_pruned_stub or task_info.base_stub
+
+
+def _download_zoo_model(model_path: str) -> str:
+    model = Zoo.load_model_from_stub(model_path)
+    return model.onnx_file.downloaded_path()
+
+
+def _overwrite_model_inputs(
+    path: str,
+    max_length: int = 128,
+) -> Tuple[str, List[str], Optional[NamedTemporaryFile]]:
+    # overwrite input shapes
+    model = onnx.load(path)
+    initializer_input_names = set([node.name for node in model.graph.initializer])
+    external_inputs = [
+        inp for inp in model.graph.input if inp.name not in initializer_input_names
+    ]
+    input_names = []
+    for external_input in external_inputs:
+        external_input.type.tensor_type.shape.dim[0].dim_value = 1
+        external_input.type.tensor_type.shape.dim[1].dim_value = max_length
+        input_names.append(external_input.name)
+
+    # Save modified model
+    tmp_file = NamedTemporaryFile()  # file will be deleted after program exit
+    onnx.save(model, tmp_file.name)
+
+    return tmp_file.name, input_names, tmp_file
+
+
+def _create_model(
+    model_path: str,
+    engine_type: str,
+    num_cores: Optional[int],
+    num_sockets: Optional[int],
+) -> Tuple[Union[Engine, InferenceSession], List[str]]:
+    onnx_path, input_names, _ = _overwrite_model_inputs(model_path)
+
+    if engine_type == DEEPSPARSE_ENGINE:
+        model = compile_model(
+            onnx_path, batch_size=1, num_cores=num_cores, num_sockets=num_sockets
+        )
```

Comment:
max_length isn't piped into _overwrite_model_inputs yet

Reply:
good catch. fixed and tested